schema_based_on_diagram.sql

```diff
@@ -0,0 +1,48 @@
+CREATE DATABASE clinic;
+
+CREATE TABLE patients (
+  id SERIAL,
+  name VARCHAR(255) NOT NULL,
+  date_of_birth DATE NOT NULL,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories (
+  id integer GENERATED ALWAYS AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id integer REFERENCES patients(id),
+  status varchar(255),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoices (
+  id integer GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL NOT NULL,
+  generated_at TIMESTAMP NOT NULL, 
+  payed_at TIMESTAMP NOT NULL,
+  medical_history_id INT NOT NULL REFERENCES medical_histories(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE treatments (
+  id integer GENERATED ALWAYS AS IDENTITY,
+  type VARCHAR(255),
+  name VARCHAR(255),
+  primary key (id)
+);
+
+CREATE TABLE invoice_items (
+  id integer GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL,
+  quantity integer,
+  total_price DECIMAL,
+  invoice_id integer REFERENCES invoices(id),
+  treatment_id integer REFERENCES treatments(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_treatments (
+  medical_id integer REFERENCES medical_histories(id),
+  treatment_id integer REFERENCES treatments(id),
+  PRIMARY KEY (medical_id, treatment_id)
+);
```

Comment:
- [ ] All FOREIGN KEYS should have indexes created, e.g `CREATE INDEX ON medical_histories (patient_id);`, kindly create indexes for all foreign keys

Reply:
Thank you for the information. I didn't pay attention to this during my reading.